### DISCUSSION TITLE
Preview for Auto Tag?

### DISCUSSION THREAD

**Discussion by nmycps** [Marekkon5/onetagger#274] [Ideas]
As it's possible to see the preview of a file naming in Auto Rename, is it possible that in Auto Tag, one could see a preview of the file's tags before they're written?

**Comment by Marekkon5** [comment]
Hello, as the name suggests "Auto Tag" is fully automatic. If you wanna preview the tags, or do cherry picking use Manual Tag (in Quick Tag right click on a track -> Manual Tag, or in Edit Tags select a track and click Manual Tag on top right.)

**Comment by nmycps** [comment]
Hello again. It was better to show you directly what I was trying to convey. This is from dbPoweramp CD Ripper.
Not sure if you have ever intended this for onetagger to have or if it is beyond the scope of the app. Thank you!

<img width="551" alt="Screenshot 2023-09-08 173424" src="https://github.com/Marekkon5/onetagger/assets/61409851/abf1060a-078b-41ef-a3ff-ab432ca3648a">
<img width="1847" alt="2023-09-08" src="https://github.com/Marekkon5/onetagger/assets/61409851/c83348cb-f46a-4d3b-8a06-c5fab3213258">

**Reply by Marekkon5** [reply]
Hello, sadly I won't be implementing this in near future because of how 1T currently works:

1. Auto Tag is meant to be fully automatic. Manual Tag was intended for this purpose (displaying the tags before being written & cherry picking)
2. Currently AutoTag writes tags on every match, so:
 - To even make this possible we would have to do large refactor
 - It would be significantly slower
 - We would have to store ALL of the track information until it gets shown & written, which when tagging 10K+ tracks is not really a good idea.
 - It would add a step where we write all the tags at once, which would be a slow down + probably not really good idea on spinning drives and such.
3. If you use the multiplatform option you would get prompted on every platform, since new platform = new tags. Again less auto, more work, and the process would stop several times. 
4. If it only shows the tags (literally "preview") I don't really see the purpose?

I hope you can understand, and sorry.

**Reply by nmycps** [reply]
Thank you. It's really great to see you have taken a careful examination of the problem and have taken the time to explain it. Much appreciated for your time.
